mlrun/artifacts/llm_prompt.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tempfile
+import typing
+
+import mlrun
+import mlrun.artifacts.model as model_artifact
+from mlrun.artifacts import Artifact, ArtifactSpec
+from mlrun.utils import StorePrefix, logger
+
+MAX_PROMPT_LENGTH = 1000
+
+
+class LLMPromptArtifactSpec(ArtifactSpec):
+    _dict_fields = ArtifactSpec._dict_fields + [
+        "prompt_string",
+        "prompt_legend",
+        "generation_configuration",
+        "description",
+    ]
+
+    def __init__(
+        self,
+        target_path=None,
+        viewer=None,
+        is_inline=False,
+        format=None,
+        size=None,
+        db_key=None,
+        extra_data=None,
+        body=None,
+        unpackaging_instructions: typing.Optional[dict] = None,
+        parent_uri: typing.Optional[str] = None,
+        prompt_string: typing.Optional[str] = None,
+        prompt_path: typing.Optional[str] = None,
+        prompt_legend: typing.Optional[dict] = None,
+        generation_configuration: typing.Optional[dict] = None,
+        description: typing.Optional[str] = None,
+    ):
+        if prompt_string and prompt_path:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "cannot specify prompt_string and prompt_path together"
+            )
+
+        if prompt_string and len(prompt_string) > MAX_PROMPT_LENGTH:
+            logger.info("prompt_string is too long, creating a temp file")
+            with tempfile.NamedTemporaryFile(
+                delete=False, mode="w", suffix=".txt"
+            ) as temp_file:
+                temp_file.write(prompt_string)
+            prompt_path = temp_file.name
+            prompt_string = None
+
+        super().__init__(
+            src_path=prompt_path,
+            target_path=target_path,
+            viewer=viewer,
+            is_inline=is_inline,
+            format=format,
+            size=size,
+            db_key=db_key,
+            extra_data=extra_data,
+            body=body,
```

Comment:
If we're not getting a body, you can put here the `prompt_string`, and then you won't need to change the `get_body` implementation (or add the `self.prompt_string` parameter).